src/backend/distributed/planner/modify_planner.c

```diff
@@ -543,6 +569,74 @@ DistributedModifyShardInterval(Query *query)
 
 
 /*
+ * UseFastShardPruning returns true if the commandType is INSERT and partition method
+ * is hash or range.
+ */
+static bool
+UseFastShardPruning(CmdType commandType, char partitionMethod)
```

Comment:
This sounds more like a an action than a boolean function. Maybe FastShardPruningPossible()?